rpc/grpc/client/block_service.go

```diff
@@ -0,0 +1,103 @@
+package client
+
+import (
+	"context"
+	"fmt"
+
+	blocksvc "github.com/cometbft/cometbft/proto/tendermint/services/block/v1"
+	"github.com/cometbft/cometbft/types"
+	"github.com/cosmos/gogoproto/grpc"
+)
+
+// Block data returned by the CometBFT BlockService gRPC API.
+type Block struct {
+	BlockID types.BlockID `json:"block_id"`
+	Block   *types.Block  `json:"block"`
+}
+
+// BlockServiceClient provides block information
+type BlockServiceClient interface {
+	GetBlockByHeight(ctx context.Context, height int64) (*Block, error)
+	// GetLatestHeight provides sends the latest committed block height to the given output
+	// channel as blocks are committed.
+	GetLatestHeight(ctx context.Context) (chan int64, chan error)
+}
+
+type blockServiceClient struct {
+	client blocksvc.BlockServiceClient
+}
+
+func newBlockServiceClient(conn grpc.ClientConn) BlockServiceClient {
+	return &blockServiceClient{
+		client: blocksvc.NewBlockServiceClient(conn),
+	}
+}
+
+// GetBlockByHeight implements BlockServiceClient GetBlockByHeight
+func (c *blockServiceClient) GetBlockByHeight(ctx context.Context, height int64) (*Block, error) {
+	req := blocksvc.GetByHeightRequest{
+		Height: height,
+	}
+	res, err := c.client.GetByHeight(ctx, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert Block from proto to core type
+	block, err := types.BlockFromProto(res.Block)
+	if err != nil {
+		return nil, err
+	}
+
+	// convert BlockID from proto to core type
+	blockID, err := types.BlockIDFromProto(res.BlockId)
+	if err != nil {
+		return nil, err
+	}
+
+	response := Block{
+		BlockID: *blockID,
+		Block:   block,
+	}
+	return &response, nil
+}
+
+// GetLatestHeight implements BlockServiceClient GetLatestHeight
+func (c *blockServiceClient) GetLatestHeight(ctx context.Context) (chan int64, chan error) {
```

Comment:
I wouldn't recommend this approach of returning two channels.

Rather:
1. Only use a single channel with a result type that encapsulates both a height and an error ([example](https://stackoverflow.com/a/25142269/1156132)).
2. Allow the caller to supply a channel as you did previously. This allows the caller to control whether it's buffered/unbuffered and the channel size.